bundles/admin/admin-hierarchical-layerlist/components/Group.js

```diff
@@ -0,0 +1,258 @@
+            popup.close();
+            tool.removeClass('active');
+        });
+        returnObject.buttons.push(btnCancel);
+        var btnOk = Oskari.clazz.create('Oskari.userinterface.component.Button');
+        returnObject.buttons.push(btnOk);
+        popup.addClass('admin-hierarchical-layerlist-group');
+
+        var loc = (Oskari.getMsg('DivManazer', 'LanguageSelect').languages) ? Oskari.getMsg('DivManazer', 'LanguageSelect').languages : {};
+
+        var selectableGroup = Oskari.clazz.create('Oskari.userinterface.component.CheckboxInput');
+        selectableGroup.addClass('selectable-group');
+        selectableGroup.setTitle(me.locale('selectableGroup'));
+        selectableGroup.setChecked(!!opts.selectable);
+
+        btnOk.addClass('add');
+        var btnOkLocale = (id) ? me.locale('buttons.update') : me.locale('buttons.add');
+        btnOk.setTitle(btnOkLocale);
+        btnOk.setHandler(function() {
+            var data = {
+                locales: {},
+                selectable: selectableGroup.isChecked(),
+                parentId: parentId,
+                id: id
+            };
+            var hasValidLocales = true;
+            popup.getJqueryContent().find('.oskari-textinput.group-name').each(function() {
+                var el = jQuery(this);
+
+                var value = el.find('input').val().trim();
+                data.locales[el.attr('data-locale')] = value;
+                if (value.length < 4) {
+                    hasValidLocales = false;
+                    el.find('input').addClass('error');
+                } else {
+                    el.find('input').removeClass('error');
+                }
+            });
+
+            if (Object.keys(data.locales).length === Oskari.getSupportedLanguages().length && hasValidLocales) {
+                me._saveGroup(data, popup, opts.type, tool);
+            } else {
+                errorDialog.show(me.locale('errors.groupname.title'), me.locale('errors.groupname.message'));
+                errorDialog.fadeout();
+            }
+
+        });
+
+        var message = jQuery('<div class="group-names"></div>');
+        returnObject.message = message;
+
+        // locale inputs
+        var supportedLocales = Oskari.getSupportedLanguages();
+        supportedLocales.forEach(function(locale) {
+            var input = Oskari.clazz.create('Oskari.userinterface.component.TextInput');
+            input.setTitle(me.locale('groupTitles.localePrefix') + ' ' + (loc[locale] || locale));
+            var value = (opts.locale && opts.locale[locale]) ? opts.locale[locale] : null;
+            if (value) {
+                input.setValue(value);
+            }
+            input.addClass('group-name');
+            var el = jQuery(input.getElement());
+            el.find('input').bind('keyup', function() {
+                var inputEl = jQuery(this);
+                var value = inputEl.val().trim();
+                if (value.length < 4) {
```

Comment:
Same here with the 4 character requirement. If one character isn't enough move the magic number 4 to a constant so it isn't repeated in multiple places (easier to change afterwards and find the places it affects).